### ISSUE TITLE
Handling of input transformation

### ISSUE THREAD

**Issue by adriangb** [adriangb/sklearn_keras_wrap#12]
The previous wrappers that lived in the tensorflow repo automatically converted from object or ordinal targets to one-hot encoded targets if the loss function was categorical crossentropy. We are doing the same here, but given that we now have access to OneHotEncoder and OrdinalEncoder I think it would be good to move away from numpy indexing stufff to a cleaner implementation.
